airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3Consumer.java

```diff
@@ -85,8 +90,20 @@ protected void acceptTracked(final AirbyteMessage airbyteMessage) throws Excepti
 
   @Override
   protected void close(final boolean hasFailed) throws Exception {
-    for (final DestinationFileWriter handler : streamNameAndNamespaceToWriters.values()) {
-      handler.close(hasFailed);
+    Optional<Exception> exceptionsThrown = Optional.empty();
```

Comment:
Different handle closure may fail due to different reasons. This variable is only able to store one exception at most. Should it be a list, and when empty, throw one exception including all the exception messages? The trade-off is that the aggregated exception won't include the stack trace. But I think that throwing out the exception message is more important than knowing all the stack trace.